k-distribution/include/kframework/builtin/domains.md

```diff
@@ -930,7 +930,7 @@ You can:
 ```k
   syntax Int ::= "~Int" Int                     [function, klabel(~Int_), symbol, functional, latex(\mathop{\sim_{\scriptstyle\it Int}}{#1}), hook(INT.not), smtlib(notInt)]
                > left:
-                 Int "^Int" Int                 [function, klabel(_^Int_), symbol, left, smt-hook(^), latex({#1}\mathrel{{\char`\^}_{\!\scriptstyle\it Int}}{#2}), hook(INT.pow)]
+                 Int "^Int" Int                 [function, klabel(_^Int_), symbol, left, smt-hook((to_int (^ #1 #2))), latex({#1}\mathrel{{\char`\^}_{\!\scriptstyle\it Int}}{#2}), hook(INT.pow)]
```

Comment:
We need to not use this encoding if the second argument is `<Int 0`. Because this is declared in a different module than the rule you added below, we are not guaranteed that a user will have the below simplification rule apply prior to sending to SMT.

If the second argument is `<Int 0`, then the overall result should be `0`. We could encode this to Z3 as `(ite (< #2 0) (0) (^ #1 #2))`.

Reply:
We also need to check that `to_int` is _truncating_, not _rounding_.